tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Models;
+using ModelContextProtocol.Server;
+using Azure.Sdk.Tools.Cli.Services.Update;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Tools;
+
+[McpServerToolType, Description("Update customized SDK code after TypeSpec regeneration: creates a new generation, diffs old vs new generated code, maps API changes to impacted customization files, applies patches.")]
+public class TspClientUpdateTool : MCPTool
+{
+    private readonly ILogger<TspClientUpdateTool> logger;
+    private readonly IOutputHelper output;
+    private readonly Func<string, IUpdateLanguageService> languageServiceFactory;
+    private readonly Argument<string> specPathArg = new(name: "spec-path", description: "Path to the .tsp specification file") { Arity = ArgumentArity.ExactlyOne };
+    private readonly Option<string?> newGenOpt = new(["--new-gen"], () => "./tmpgen", "Directory for regenerated TypeSpec output (optional)");
+
+    public TspClientUpdateTool(ILogger<TspClientUpdateTool> logger, IOutputHelper output, Func<string, IUpdateLanguageService> languageServiceFactory)
+    {
+        this.logger = logger;
+        this.output = output;
+        this.languageServiceFactory = languageServiceFactory;
+        CommandHierarchy = [ SharedCommandGroups.TypeSpec ];
+    }
+
+    public override Command GetCommand()
+    {
+        var cmd = new Command("customized-update",
+            description: "Update customized TypeSpec-generated client code. Runs the full pipeline by default: regenerate -> diff -> map -> propose -> apply");
+        cmd.AddArgument(specPathArg);
+        cmd.AddOption(SharedOptions.PackagePath);
+        cmd.AddOption(newGenOpt);
+        cmd.SetHandler(async ctx => await HandleUnified(ctx, ctx.GetCancellationToken()));
+        return cmd;
+    }
+
+    public override Task HandleCommand(InvocationContext ctx, CancellationToken ct) => Task.CompletedTask;
+
+    [McpServerTool(Name = "azsdk_tsp_update"), Description("Update pipeline")] 
+    public async Task<TspClientUpdateResponse> UnifiedUpdateAsync(string specPath, IUpdateLanguageService languageService, CancellationToken ct = default)
+    {
+        try
+        {
+            if (string.IsNullOrWhiteSpace(specPath))
+            {
+                return new TspClientUpdateResponse { ResponseError = "Spec path required", ErrorCode = "MissingSpec" };
+            }
+            var session = new UpdateSessionState { SpecPath = specPath };
```

Comment:
Does this tool update `typespec`? Should we track `packagePath` in the session instead of using spec if the answer is no?

Reply:
It doesn’t update the .tsp spec itself; it only consumes the spec to regenerate client code and then updates customization.